### PR TITLE
Incorrect Signature algorithms presented by Client Authenticator

### LINKED ISSUE #15853: Incorrect Signature algorithms presented by Client Authenticator
### Area

oidc

## Issue:

Presented signature algorithms for clients using client Client authenticator being "Signed JWT" or "Signed JWT with client Secret" are not correct

### Signed JWT client 

Doc states

```
Signed JWT is "Signed Json Web Token".

When choosing this credential type you will have to also generate a private key and certificate for the client in the tab Keys. The private key will be used to sign the JWT, while the certificate is used by the server to verify the signature.
```

Keycloak behaviour

1. Go to credentials
2. Select client authenticator "Signed Jwt"
3. Go to Signature algorithm

The proposed list of signature algorithm iss:
```
ES256
ES384
ES512
HS256
HS384
HS512
PS256
PS384
PS512
RS256
RS384
RS512
```

Client has his private key registered within a keystore. It means that asymmetric key algorithm are needs to be used.

The value HS256 HS384 HS512 are applying to HMAC which is a a symmetric algorithm. 

The client needs to use a asymmetric algorithm.

It is providing more confusion than anything else, and those values should be removed.

 

The list of proposed algorithms should be 
```
ES256
ES384
ES512
PS384
PS512
RS256
RS384
RS512
```

#### Signed JWT with client secret

The doc states

```
If you select this option, you can use a JWT signed by client secret instead of the private key.

The client secret will be used to sign the JWT by the client.
```

Keycloak behaviour:

1. Go to credentials
2. Select client authenticator "Signed Jwt with client credentials"
3. Go to Signature algorithm

The proposed list of signature algorithm proposes:
```
ES256
ES384
ES512
HS256
HS384
HS512
PS256
PS384
PS512
RS256
RS384
RS512
```

Here we are dealing with symmetric key encryption, where is no Keystore.

Even more, there is reference within the doc, or corresponding client authenticator panel to any JWKS URL.

Also the OIDC specification explicitly mention the HMAC based algorithm in the section for `client_secret_jwt` : https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication

Therefore the proposed list should only contains the symmetric algorithm: 
```
HS256
HS384
HS512
```

### Version

20.0.0

### PR DESCRIPTION
**Pull request by mposolda**
closes #15853

### CODE REVIEW

File: js/apps/admin-ui/src/clients/credentials/SignedJWT.tsx [hunk 1]
```diff
@@ -13,11 +13,15 @@ import { HelpItem } from "ui-shared";
...
+};
+
+export const SignedJWT = ({ clientAuthenticatorType }: SignedJWTProps) => {
   const { control } = useFormContext();
-  const providers = sortProviders(
-    useServerInfo().providers!.clientSignature.providers
-  );
+  const providers =  clientAuthenticatorType === "client-jwt"
+    ? (useServerInfo().cryptoInfo?.clientSignatureAsymmetricAlgorithms ?? [])
+    : (useServerInfo().cryptoInfo?.clientSignatureSymmetricAlgorithms ?? []);
```
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/credentials/SignedJWT.tsx:24]: Although this does not result in errors, hooks [should not be called conditionally](https://react.dev/warnings/invalid-hook-call-warning). Instead you can store the value in side of a variable outside of these conditions:

```suggestion
  const { cryptoInfo } = useServerInfo();
  const providers =
    clientAuthenticatorType === "client-jwt"
      ? cryptoInfo?.clientSignatureAsymmetricAlgorithms ?? []
      : cryptoInfo?.clientSignatureSymmetricAlgorithms ?? [];`
``

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — DISMISSED** [review]
Changes on the font-end side of things look good.

**Review by tnorimat — APPROVED** [review]
LGTM

**Comment by tnorimat** [comment]
@mposolda As one optional comment,  whether are UI tests needed or not.

**Comment by mposolda** [comment]
@jonkoops @tnorimat Thanks!

**Comment by mposolda** [comment]
> @mposolda As one optional comment, whether are UI tests needed or not.

@tnorimat Thanks for the comment. It seems that `ServerInfoTest` is ok based on the approval from Jon from our UI team.

**Pull request merged**